src/app/core/services/filters.service.ts

```diff
@@ -38,33 +40,47 @@ export class FiltersService {
   }
 
   updateFilters(newFilters: Partial<Filter>): void {
-    let nextDropdownFilter: Filter = {
+    let nextFilter: Filter = {
       ...this.filter$.value,
       ...newFilters
     };
 
-    nextDropdownFilter = this._validateFilter(nextDropdownFilter);
+    nextFilter = this._validateFilter(nextFilter);
 
-    this.filter$.next(nextDropdownFilter);
+    this.filter$.next(nextFilter);
   }
 
+  sanitizeLabels(allLabels: SimpleLabel[]) {
+    const allLabelsSet = new Set(allLabels.map((label) => label.name));
+
+    const newHiddenLabels: Set<string> = new Set();
+    for (const hiddenLabel of this.filter$.value.hiddenLabels) {
+      if (allLabelsSet.has(hiddenLabel)) {
+        newHiddenLabels.add(hiddenLabel);
+      }
+    }
+
+    const newLabels = this.filter$.value.labels.filter((label) => allLabelsSet.has(label));
+
+    this.updateFilters({ labels: newLabels, hiddenLabels: newHiddenLabels });
+  }
   /**
    * Changes type to a valid, default value when an incompatible combination of type and status is encountered.
    */
-  updateTypePairing(dropdownFilter: Filter): Filter {
-    if (dropdownFilter.status === 'merged') {
-      dropdownFilter.type = 'pullrequest';
+  updateTypePairing(filter: Filter): Filter {
+    if (filter.status === 'merged') {
+      filter.type = 'pullrequest';
     }
-    return dropdownFilter;
+    return filter;
   }
 
   /**
    * Changes status to a valid, default value when an incompatible combination of type and status is encountered.
    */
-  updateStatusPairing(dropdownFilter: Filter): Filter {
-    if (dropdownFilter.status === 'merged' && dropdownFilter.type === 'issue') {
-      dropdownFilter.status = 'all';
+  updateStatusPairing(filter: Filter): Filter {
+    if (filter.status === 'merged' && filter.type === 'issue') {
+      filter.status = 'all';
     }
-    return dropdownFilter;
+    return filter;
```

Comment:
In my opinion, since the ``filter`` (previously ``dropdownFilter``) variable now represents all the filters instead of only the dropdown filters, they should be named appropriately for consistency.
Is there a reason we shouldn't change the names?